<ui version="4.0" >
 <author>Roman Kamyk</author>
 <comment></comment>
 <exportmacro></exportmacro>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>945</width>
    <height>952</height>
   </rect>
  </property>
  <property name="sizePolicy" >
   <sizepolicy>
    <hsizetype>7</hsizetype>
    <vsizetype>7</vsizetype>
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle" >
   <string>Detection</string>
  </property>
  <widget class="QWidget" name="centralwidget" >
   <layout class="QVBoxLayout" >
    <property name="margin" >
     <number>9</number>
    </property>
    <property name="spacing" >
     <number>6</number>
    </property>
    <item>
     <widget class="QGroupBox" name="ControlGB" >
      <property name="sizePolicy" >
       <sizepolicy>
        <hsizetype>7</hsizetype>
        <vsizetype>1</vsizetype>
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize" >
       <size>
        <width>16777215</width>
        <height>140</height>
       </size>
      </property>
      <property name="title" >
       <string>Control</string>
      </property>
      <layout class="QHBoxLayout" >
       <property name="margin" >
        <number>2</number>
       </property>
       <property name="spacing" >
        <number>5</number>
       </property>
       <item>
        <layout class="QGridLayout" >
         <property name="margin" >
          <number>0</number>
         </property>
         <property name="spacing" >
          <number>6</number>
         </property>
         <item row="0" column="3" >
          <widget class="QPushButton" name="loadFileBtn" >
           <property name="text" >
            <string>Load file</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2" >
          <widget class="QToolButton" name="chooseLearningBtn" >
           <property name="text" >
            <string>...</string>
           </property>
           <property name="icon" >
            <iconset resource="Images.qrc" >:/img/img/folder_sound.png</iconset>
           </property>
          </widget>
         </item>
         <item row="1" column="0" >
          <widget class="QLabel" name="label" >
           <property name="text" >
            <string>Learning directory:</string>
           </property>
           <property name="buddy" >
            <cstring>learningDirectoryEdt</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1" >
          <widget class="QLineEdit" name="fileNameEdt" >
           <property name="sizePolicy" >
            <sizepolicy>
             <hsizetype>7</hsizetype>
             <vsizetype>0</vsizetype>
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="1" column="1" >
          <widget class="QLineEdit" name="learningDirectoryEdt" >
           <property name="text" >
            <string>./samples/</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2" >
          <widget class="QToolButton" name="chooseFileBtn" >
           <property name="text" >
            <string>...</string>
           </property>
           <property name="icon" >
            <iconset resource="Images.qrc" >:/img/img/player_eject_flat.png</iconset>
           </property>
          </widget>
         </item>
         <item row="1" column="3" >
          <widget class="QPushButton" name="loadLearningBtn" >
           <property name="text" >
            <string>Load learning set</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0" >
          <widget class="QLabel" name="label" >
           <property name="text" >
            <string>Filename:</string>
           </property>
           <property name="buddy" >
            <cstring>fileNameEdt</cstring>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar" >
         <property name="sizePolicy" >
          <sizepolicy>
           <hsizetype>5</hsizetype>
           <vsizetype>0</vsizetype>
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize" >
          <size>
           <width>100</width>
           <height>0</height>
          </size>
         </property>
         <property name="value" >
          <number>0</number>
         </property>
         <property name="textVisible" >
          <bool>true</bool>
         </property>
         <property name="orientation" >
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget" >
      <layout class="QHBoxLayout" >
       <property name="margin" >
        <number>1</number>
       </property>
       <property name="spacing" >
        <number>5</number>
       </property>
       <item>
        <widget class="QGroupBox" name="TimeDomainGB" >
         <property name="title" >
          <string>Time domain</string>
         </property>
         <layout class="QVBoxLayout" >
          <property name="margin" >
           <number>2</number>
          </property>
          <property name="spacing" >
           <number>1</number>
          </property>
          <item>
           <widget class="QGroupBox" name="audioSignalGB" >
            <property name="sizePolicy" >
             <sizepolicy>
              <hsizetype>7</hsizetype>
              <vsizetype>7</vsizetype>
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title" >
             <string>Audio signal</string>
            </property>
            <layout class="QHBoxLayout" >
             <property name="margin" >
              <number>3</number>
             </property>
             <property name="spacing" >
              <number>0</number>
             </property>
             <item>
              <widget class="SignalDraw" name="audioSignalDraw" >
               <property name="toolTip" >
                <string>Draws audio signal</string>
               </property>
               <property name="whatsThis" >
                <string>Draws signal.</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="FilterGB" >
            <property name="sizePolicy" >
             <sizepolicy>
              <hsizetype>7</hsizetype>
              <vsizetype>7</vsizetype>
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title" >
             <string>Digitally filtered signal</string>
            </property>
            <layout class="QVBoxLayout" >
             <property name="margin" >
              <number>3</number>
             </property>
             <property name="spacing" >
              <number>0</number>
             </property>
             <item>
              <widget class="SignalDraw" name="filteredDraw" >
               <property name="toolTip" >
                <string>Draws audio signal</string>
               </property>
               <property name="whatsThis" >
                <string>Draws signal.</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox" >
            <property name="title" >
             <string>Signal energy</string>
            </property>
            <layout class="QHBoxLayout" >
             <property name="margin" >
              <number>3</number>
             </property>
             <property name="spacing" >
              <number>0</number>
             </property>
             <item>
              <widget class="EnergyDrawWidget" name="energyDraw" >
               <property name="toolTip" >
                <string>Draws audio signal energy</string>
               </property>
               <property name="whatsThis" >
                <string>Draws signal's energy.</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="SegmentGB" >
            <property name="sizePolicy" >
             <sizepolicy>
              <hsizetype>7</hsizetype>
              <vsizetype>7</vsizetype>
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title" >
             <string>Selected segment</string>
            </property>
            <layout class="QVBoxLayout" >
             <property name="margin" >
              <number>3</number>
             </property>
             <property name="spacing" >
              <number>0</number>
             </property>
             <item>
              <widget class="SignalDraw" name="segmentDraw" >
               <property name="toolTip" >
                <string>Draws audio signal</string>
               </property>
               <property name="whatsThis" >
                <string>Draws signal.</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="FrequencyDomainGB" >
         <property name="title" >
          <string>Frequency domain</string>
         </property>
         <layout class="QHBoxLayout" >
          <property name="margin" >
           <number>2</number>
          </property>
          <property name="spacing" >
           <number>1</number>
          </property>
          <item>
           <layout class="QVBoxLayout" >
            <property name="margin" >
             <number>0</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item>
             <widget class="QGroupBox" name="SpectrogramGB" >
              <property name="sizePolicy" >
               <sizepolicy>
                <hsizetype>7</hsizetype>
                <vsizetype>7</vsizetype>
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title" >
               <string>Spectrogram</string>
              </property>
              <layout class="QVBoxLayout" >
               <property name="margin" >
                <number>3</number>
               </property>
               <property name="spacing" >
                <number>0</number>
               </property>
               <item>
                <widget class="SpectroDraw" name="spectrogram" >
                 <property name="toolTip" >
                  <string>Draws audio signal spectrogram</string>
                 </property>
                 <property name="whatsThis" >
                  <string>Draws signal.</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="NormalizedGB" >
              <property name="sizePolicy" >
               <sizepolicy>
                <hsizetype>7</hsizetype>
                <vsizetype>7</vsizetype>
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title" >
               <string>Normalized spectrogram</string>
              </property>
              <layout class="QVBoxLayout" >
               <property name="margin" >
                <number>3</number>
               </property>
               <property name="spacing" >
                <number>0</number>
               </property>
               <item>
                <widget class="SpectroDraw" name="normalizedSpectrogram" >
                 <property name="toolTip" >
                  <string>Draws audio signal spectrogram</string>
                 </property>
                 <property name="whatsThis" >
                  <string>Draws signal.</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="BestMatchGB" >
              <property name="sizePolicy" >
               <sizepolicy>
                <hsizetype>7</hsizetype>
                <vsizetype>7</vsizetype>
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title" >
               <string>Best match sample</string>
              </property>
              <layout class="QHBoxLayout" >
               <property name="margin" >
                <number>3</number>
               </property>
               <property name="spacing" >
                <number>0</number>
               </property>
               <item>
                <widget class="SpectroDraw" name="bestMatchSpectrogram" >
                 <property name="toolTip" >
                  <string>Draws audio signal spectrogram</string>
                 </property>
                 <property name="whatsThis" >
                  <string>Draws signal.</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>0</y>
     <width>945</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuPlay" >
    <property name="title" >
     <string>Play</string>
    </property>
    <addaction name="actionPlaySelection" />
    <addaction name="actionStop" />
    <addaction name="actionRecord" />
    <addaction name="separator" />
    <addaction name="actionMakeSpectrogram" />
    <addaction name="actionCompare" />
   </widget>
   <widget class="QMenu" name="menuEdit" >
    <property name="title" >
     <string>Edit</string>
    </property>
    <addaction name="actionZoomIn" />
    <addaction name="actionZoomOut" />
    <addaction name="actionZoom11" />
    <addaction name="separator" />
    <addaction name="actionSettings" />
   </widget>
   <widget class="QMenu" name="menuFile" >
    <property name="title" >
     <string>File</string>
    </property>
    <addaction name="actionOpen" />
    <addaction name="actionLoadLerningSet" />
    <addaction name="actionReadLearning" />
    <addaction name="actionSaveSelection" />
    <addaction name="separator" />
    <addaction name="actionExit" />
   </widget>
   <addaction name="menuFile" />
   <addaction name="menuEdit" />
   <addaction name="menuPlay" />
  </widget>
  <widget class="QStatusBar" name="statusbar" />
  <widget class="QToolBar" name="toolBar" >
   <property name="orientation" >
    <enum>Qt::Horizontal</enum>
   </property>
   <attribute name="toolBarArea" >
    <number>4</number>
   </attribute>
   <addaction name="actionOpen" />
   <addaction name="actionLoadLerningSet" />
   <addaction name="actionReadLearning" />
   <addaction name="actionSaveSelection" />
   <addaction name="separator" />
   <addaction name="actionPlaySelection" />
   <addaction name="actionRecord" />
   <addaction name="actionStop" />
   <addaction name="actionMakeSpectrogram" />
   <addaction name="actionCompare" />
   <addaction name="separator" />
   <addaction name="actionZoomIn" />
   <addaction name="actionZoomOut" />
   <addaction name="actionZoom11" />
   <addaction name="separator" />
   <addaction name="actionSettings" />
   <addaction name="actionOpenLearningEditor" />
  </widget>
  <action name="actionOpen" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/player_eject_flat.png</iconset>
   </property>
   <property name="text" >
    <string>Open sample</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionPlaySelection" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/player_play_flat.png</iconset>
   </property>
   <property name="text" >
    <string>Play Selection</string>
   </property>
   <property name="shortcut" >
    <string>A</string>
   </property>
  </action>
  <action name="actionStop" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/player_stop_flat.png</iconset>
   </property>
   <property name="text" >
    <string>Stop</string>
   </property>
   <property name="shortcut" >
    <string>S</string>
   </property>
  </action>
  <action name="actionLoadLerningSet" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/folder_sound.png</iconset>
   </property>
   <property name="text" >
    <string>Load lerning set</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionRecord" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/player_record.png</iconset>
   </property>
   <property name="text" >
    <string>Record</string>
   </property>
   <property name="shortcut" >
    <string>D</string>
   </property>
  </action>
  <action name="actionExit" >
   <property name="text" >
    <string>Exit</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionMakeSpectrogram" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/kview.png</iconset>
   </property>
   <property name="text" >
    <string>Make Spectrogram</string>
   </property>
   <property name="shortcut" >
    <string>M</string>
   </property>
  </action>
  <action name="actionZoomIn" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/viewmag+.png</iconset>
   </property>
   <property name="text" >
    <string>Zoom in</string>
   </property>
   <property name="shortcut" >
    <string>+</string>
   </property>
  </action>
  <action name="actionZoom11" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/viewmag1.png</iconset>
   </property>
   <property name="text" >
    <string>Zoom 1:1</string>
   </property>
   <property name="shortcut" >
    <string>Z</string>
   </property>
  </action>
  <action name="actionZoomOut" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/viewmag-.png</iconset>
   </property>
   <property name="text" >
    <string>Zoom out</string>
   </property>
   <property name="shortcut" >
    <string>-</string>
   </property>
  </action>
  <action name="actionCompare" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/autodiff.png</iconset>
   </property>
   <property name="text" >
    <string>Compare</string>
   </property>
   <property name="shortcut" >
    <string>C</string>
   </property>
  </action>
  <action name="actionSettings" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/options.png</iconset>
   </property>
   <property name="text" >
    <string>Settings</string>
   </property>
   <property name="iconText" >
    <string>Settings</string>
   </property>
   <property name="toolTip" >
    <string>Settings</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionSaveSelection" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/backup_section.png</iconset>
   </property>
   <property name="text" >
    <string>Save selection</string>
   </property>
   <property name="shortcut" >
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionOpenLearningEditor" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/artsenvironment.png</iconset>
   </property>
   <property name="text" >
    <string>Open learning set editor</string>
   </property>
  </action>
  <action name="actionReadLearning" >
   <property name="icon" >
    <iconset resource="Images.qrc" >:/img/img/folder_binary.png</iconset>
   </property>
   <property name="text" >
    <string>Read learning set from file</string>
   </property>
  </action>
 </widget>
 <pixmapfunction></pixmapfunction>
 <customwidgets>
  <customwidget>
   <class>EnergyDrawWidget</class>
   <extends>QWidget</extends>
   <header>EnergyDrawWidget.hxx</header>
   <container>0</container>
   <pixmap></pixmap>
  </customwidget>
  <customwidget>
   <class>SignalDraw</class>
   <extends>QWidget</extends>
   <header>SignalDraw.hxx</header>
   <container>0</container>
   <pixmap></pixmap>
  </customwidget>
  <customwidget>
   <class>SpectroDraw</class>
   <extends>QWidget</extends>
   <header>SpectroDraw.hxx</header>
   <container>0</container>
   <pixmap></pixmap>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>chooseFileBtn</tabstop>
  <tabstop>chooseLearningBtn</tabstop>
  <tabstop>loadFileBtn</tabstop>
  <tabstop>loadLearningBtn</tabstop>
  <tabstop>fileNameEdt</tabstop>
  <tabstop>learningDirectoryEdt</tabstop>
 </tabstops>
 <resources>
  <include location="Images.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>fileNameEdt</sender>
   <signal>returnPressed()</signal>
   <receiver>loadFileBtn</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>544</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel" >
     <x>825</x>
     <y>97</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>learningDirectoryEdt</sender>
   <signal>returnPressed()</signal>
   <receiver>loadLearningBtn</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>671</x>
     <y>134</y>
    </hint>
    <hint type="destinationlabel" >
     <x>825</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
